### PR TITLE
Makes cat cyborgs more cat like

### PR DESCRIPTION
**Pull request by pixelkitty286**
## About The Pull Request

This PR makes cat cyborgs more cat like by adding cat like grace.

It also adds a very quiet servo like sound to cat cyborgs when they move wile rewriting how unique cyborgs such as tall borgs have their walking sounds applied.

Also adds a small sprite animation when cat cyborgs are dead.

## Why It's Good For The Game

Better death sprites for cat cyborgs so they do NOT just get forgotten when they die because they blend too well with the environment of the station.

The quiet servo sounds add a little flavor that they are a smaller sized cyborg unlike their bigger counter part cyborgs.

Cat like grace was added because why not and it fits the flavor of a feline like robot only down side is anything that hates small sized mobs will attack.

## Proof Of Testing

https://github.com/user-attachments/assets/08ce2104-a11d-43ea-aea6-f5d4a6ce26a2

dreammaker_HL4zDZJlyy

## Changelog

:cl:
add: cat like grace to lightweight chassis and small quad borg walking sounds
image: catborg death sprites
code: changed how cyborgs get their walking sounds.
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/borgs/sprites/felibot_all.dmi
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/borgs/sprites/felibot_all.dmi:1]: +++

**Review by projectkepler-ru — APPROVED** [review]
sprite is f i n e

File: modular_zubbers/code/modules/borgs/code/robot_model.dm [hunk 1]
```diff
@@ -25,9 +25,31 @@
...
+// To load the correct walking sounds with out removing them
+/obj/item/robot_model/proc/update_footsteps()
+	var/mob/living/silicon/robot/cyborg = robot || loc
+	if (!istype(robot))
+		return
+	if (model_features && ((TRAIT_R_SQUADRUPED in model_features) || (TRAIT_R_TALL in model_features) || (TRAIT_R_WIDE in model_features)))
+		if (model_features && (TRAIT_R_SQUADRUPED in model_features))
+			cyborg.AddElement(/datum/element/footstep, FOOTSTEP_ROBOT_SMALL, 6, -6, sound_vary = TRUE)
+		if (model_features && (TRAIT_R_TALL in model_features))
+			cyborg.AddElement(/datum/element/footstep, FOOTSTEP_MOB_SHOE, 2, -6, sound_vary = TRUE)
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/borgs/code/robot_model.dm:45]: These should probably be else ifs just so we don't try to add footsteps twice

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/borgs/code/robot_model.dm (hunk 1 above)
Comment (pixelkitty286) [reply modular_zubbers/code/modules/borgs/code/robot_model.dm:45]: I tried it and it breaks when doing such and applies both.

**Comment by lessthnthree** [comment]
add catborgs to the laser pointer check

**Comment by pixelkitty286** [comment]
Bit on hold on this waiting on something...

**Comment by Majkl-J** [comment]
> Bit on hold on this waiting on something...

I keep forgetting to review this holdon

**Comment by pixelkitty286** [comment]
> > Bit on hold on this waiting on something...
> 
> I keep forgetting to review this holdon

need that laser pointer thing on bubber to finish this because @lessthnthree had a good suggestion

**Comment by Majkl-J** [comment]
> > > Bit on hold on this waiting on something...
> > 
> > 
> > I keep forgetting to review this holdon
> 
> need that laser pointer thing on bubber to finish this because @lessthnthree had a good suggestion

The fix is on its way but tbh you should put it in another PR after this one

File: modular_zubbers/code/modules/borgs/code/robot_model.dm [hunk 2]
```diff
@@ -25,9 +25,33 @@
...
+		if (TRAIT_R_SQUADRUPED in model_features)
+			cyborg.AddElement(/datum/element/footstep, FOOTSTEP_ROBOT_SMALL, 6, -6, sound_vary = TRUE)
+		else
+			cyborg.RemoveElement(/datum/element/footstep, FOOTSTEP_ROBOT_SMALL, 6, -6, sound_vary = TRUE)
+
+		if (TRAIT_R_TALL in model_features)
+			cyborg.AddElement(/datum/element/footstep, FOOTSTEP_MOB_SHOE, 2, -6, sound_vary = TRUE)
+		else
+			cyborg.RemoveElement(/datum/element/footstep, FOOTSTEP_MOB_SHOE, 2, -6, sound_vary = TRUE)
+
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/borgs/code/robot_model.dm:53]: Lemme test if I didnt fuck this up and then I'll merge.

**Review by Majkl-J — APPROVED** [review]

**Comment by pixelkitty286** [comment]
I'll make another PR on it did you run into the issue i did on it?

**Pull request merged**